apps/aecore/src/aec_fork_block_settings.erl

```diff
@@ -0,0 +1,76 @@
+-module(aec_fork_block_settings).
+
+-include_lib("aecontract/include/hard_forks.hrl").
+
+-export([dir/1,
+         genesis_accounts/0,
+         minerva_accounts/0]).
+
+-define(GENESIS_DIR, ".genesis").
+-define(MINERVA_DIR, ".minerva").
+
+
+-spec dir(aec_hard_forks:protocol_vsn()) -> string().
+dir(ProtocolVsn) ->
+    Dir =
+        case ProtocolVsn of
+            ?ROMA_PROTOCOL_VSN    -> ?GENESIS_DIR;
+            ?MINERVA_PROTOCOL_VSN -> ?MINERVA_DIR
+        end,
+    filename:join(aeu_env:data_dir(aecore), Dir).
+
+-spec genesis_accounts() -> list().
+genesis_accounts() -> preset_accounts(?ROMA_PROTOCOL_VSN,
+                                      genesis_accounts_file_missing).
+
+-spec minerva_accounts() -> list().
+minerva_accounts() -> preset_accounts(?MINERVA_PROTOCOL_VSN,
+                                      minerva_accounts_file_missing).
+
+-spec preset_accounts(aec_hard_forks:protocol_vsn(), atom()) -> list().
+preset_accounts(Release, ErrorMsg) ->
+    case read_presets(Release) of
+        {error, {_Err, PresetAccountsFile}} ->
+            % no setup, no preset accounts
+            erlang:error({ErrorMsg, PresetAccountsFile});
+        {ok, JSONData} ->
+            DecodedData =
+                try jsx:decode(JSONData) of
+                    %% JSX decodes an empty json object as [{}]:
+                    %% [{}] = jsx:decode(<<"{}">>).
+                    [{}] -> [];
+                    L when is_list(L) -> L
+                catch
+                  error:_ ->
+                    erlang:error(invalid_accounts_json)
+                end,
+            Accounts =
+                lists:map(
+                    fun({EncodedPubKey, Amt}) ->
+                        {ok, PubKey} = aehttp_api_encoder:safe_decode(account_pubkey, EncodedPubKey),
+                        {PubKey, Amt}
+                    end,
+                    DecodedData),
+            % ensure deterministic ordering of accounts
+            lists:keysort(1, Accounts)
```

Comment:
Should we also check for duplicates?

Reply:
Depends on how the scraping script is written?! I would guess some people to have migrated in several batches (I think you were even encouraged to try with a smaller amount first?!).

Reply:
Scrapping script does accumulate balances if a certain account has migrated more than once.
Besides that duplicating accounts in the JSON:
* for genesis migration it could crash while creating an account twice. This did not happen :)
* for `minerva` if an account is not present it is created and if it is present - it is updated, so duplicating accounts even if were present, would not impose a problem